documentation.md

```diff
@@ -0,0 +1,669 @@
+- Internal identifier (_iid_): Contains the identifier that will be used to identify the extended cell where to release some response.
+- Address (_addr_): Contains the base address of the address request.
+  It is used to estimate the delay caused by the requests contained in the burst.
+- Burst size (_burst_size_): Contains the burst size of the address request.
+  It is used to calculate the address of the subsequent.
+  It may also be used in the future for improved delay estimation.
+- Data valid (_data_v_): Contains one bit per elementary burst request.
+  Each bit is unset iff the corresponding elementary burst request is awaited.
+- Memory pending (_mem_pending_): Contains one bit per elementary burst request.
+  Each bit is set iff there is currently a simulated memory operation pending.
+- Memory done (_mem_done_): Contains one bit per elementary burst request.
+  Each bit is set iff the corresponding simulated memory operation is done, or this is an entry beyond the burst length.
+
+_Read slots_ are defined identically, except that:
+
+- There is no _data_v_ field, since elementary burst read requests arrive simultaneously with the read address request.
+- There internal identifier for read and write address requests do not need to be of equal width, which depends on the maximal number of pending requests.
+
+The burst length does not need to be stored explicitly, as the _data_v_ and _mem_done_ bit arrays are set accordingly during the slot allocation.
+
+#### Delay estimation
+
+The delay estimation is performed using a decrementing counter (_rank_delay_cnt_) per rank (currently, only one rank is supported).
+When a simulated memory operation starts, the decrementing counter is set to a value corresponding to the situation:
+
+1. If this is a row hit.
+2. If this is a row miss, and there was no row in the buffer.
+3. If this is a row miss, and there was another row in the buffer.
+
+Rank states are represented by two additional elements (in addition to the decrementing counter):
+
+- _is_row_open_: Is set iff there is a row stored in the simulated row buffer.
+  Currently, this signal is only unset before the first memory request.
+- _row_start_addr_: Stores the identifier of the currently open row, if any.
+  The row identifier corresponds to all the most significant bits, that are not used for intra-row addressing.
+
+#### Release enable structures
+
+Write responses have a release enable structure made of one flip-flop per internal identifier.
+When the flip-flop is set, then the corresponding release eanble signal is propagated to the write response bank, which releases the signal if the requester is ready and if the signal is at the response head of its linked list.
+
+As read data come as a burst, and to allow the progressive release of the data in the burst, small counters are used to maintain the number of read data that can currently be released per internal identifier (_i.e._, per extended cell).
```

Comment:
Does this mean the simulated memory controller allows for read interleaving of bursts with different AXI ID?

Reply:
Yes, I added a little paragraph discussing an approximation that I made.